### PR TITLE
Get crops from DB

### PR DESCRIPTION
**Pull request by sprucely**
I was originally going to have the typeahead load crops as needed, but ended up just loading them all ahead of time based on crop type. So CropController/Search has an optional query parameter that isn't being used by the client. I don't think it hurts to leave it there.

### CODE REVIEW

File: Harvest.Web/ClientApp/src/Requests/Crops.tsx [hunk 1]
```diff
@@ -40,8 +43,29 @@ export const Crops = (props: Props) => {
...
+
+        getIsMounted() && setOptions(crops);
+      }
+    }
+    getIsMounted() && setIsSearchLoading(false);
+  };
+
   return (
-    <Typeahead
+    <Typeahead<string>
```
Comment (srkirkland) [inline Harvest.Web/ClientApp/src/Requests/Crops.tsx:68]: is this newly required?  never seen this syntax inside component before, looks different

File: Harvest.Web/ClientApp/src/Requests/Crops.tsx [hunk 2]
```diff
@@ -40,8 +43,29 @@ export const Crops = (props: Props) => {
     }
   };
 
+  const onSearch = async (query?: string) => {
+    setIsSearchLoading(true);
+
+    const response = await fetch(
+      `/Crop/Search?type=${props.cropType}${(query && `&query=${query}`) || ""}`
```
Comment (srkirkland) [inline Harvest.Web/ClientApp/src/Requests/Crops.tsx:50]: I'm ok with leaving the query param in the server method but I would rather you just remove it from here to make it more readable.

**Comment by srkirkland** [comment]
Also seems like this broke the tests

File: Harvest.Web/Controllers/CropController.cs [hunk 3]
```diff
@@ -142,5 +142,17 @@ public async Task<ActionResult> Delete(int id, Crop model)
                 return View();
             }
         }
+
+        // GET: CropController/Search?type=Row
+        public async Task<ActionResult> Search([FromQuery] string type, [FromQuery] string query = "")
```
Comment (jSylvestre) [inline Harvest.Web/Controllers/CropController.cs:147]: Can't have it here. This entire class if FM access.

**Review by jSylvestre — CHANGES_REQUESTED** [review]

File: Harvest.Web/Controllers/CropController.cs (hunk 3 above)
Comment (srkirkland) [reply Harvest.Web/Controllers/CropController.cs:147]: yea you might be able to override w/ a new `[Authorize]` attribute which changes the policy to just any authorized user and still keep it in the class, but yup it needs to be changed.

File: Harvest.Web/ClientApp/src/Requests/Crops.tsx (hunk 1 above)
Comment (sprucely) [reply Harvest.Web/ClientApp/src/Requests/Crops.tsx:68]: No, but I didn't like that options were being stored as type `string | { label: string }`. Which reminds me, I forgot to remove that union from the onSelect parameter.

File: Harvest.Web/Controllers/CropController.cs (hunk 3 above)
Comment (sprucely) [reply Harvest.Web/Controllers/CropController.cs:147]: Removed FM from class and applied to all methods but this one.

File: Harvest.Web/ClientApp/src/Requests/Crops.tsx (hunk 1 above)
Comment (sprucely) [reply Harvest.Web/ClientApp/src/Requests/Crops.tsx]: Nevermind. Just realized that the funky union type it was showing was only due to the data being passed in. Making the generic type explicit is redundant, so I removed it.

**Pull request merged**